### PR TITLE
amdgpu verbumps

### PR DESCRIPTION
**Pull request by AdelKS**
Hello @thesamesam 

I am not sure how I should handle old ebuilds: whether to start dropping them or keep

The newer versions of the driver sometimes introduce bugs, and people with old hardware may prefer older drivers. So, for now, I am going for keeping ebuilds afloat as long as they're in the upstream server https://repo.radeon.com/amdgpu/

### CODE REVIEW

**Comment by dlan17** [comment]
> Hello @thesamesam
> 
> I am not sure how I should handle old ebuilds: whether to start dropping them or keep
> 
> The newer versions of the driver sometimes introduce bugs, and people with old hardware may prefer older drivers. So, for now, I am going for keeping ebuilds afloat as long as they're in the upstream server https://repo.radeon.com/amdgpu/

it's totally sane to keep older versions if there are valid reasons, like certain hw only work with old version, so no problem

but remember to drop olds if there is severe security bugs (or any nasty things happen)..

**Comment by AdelKS** [comment]
> certain hw only work with old version, so no problem

I am not sure actually about that, I think amdgpu didn't drop any old GPU from its supported list yet 🤔 but it doesn't mean that new releases still keep them in mind and may introduce regressions from them.

> but remember to drop olds if there is severe security bugs (or any nasty things happen)..

For sure

File: media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+DESCRIPTION="AMD's closed source Advanced Media Framework (AMF) driver"
+HOMEPAGE="https://www.amd.com/en/support"
+SRC_URI="https://repo.radeon.com/amdgpu/${INTERNAL_VER}/ubuntu/pool/proprietary/a/${MY_PN}/${MY_PN}_${MY_PV_FULL}.${UBUNTU_VER}_amd64.deb -> ${P}.deb"
+
+S="${WORKDIR}"
+
+RESTRICT="bindist mirror"
+
+LICENSE="AMD-GPU-PRO-EULA"
+SLOT="0"
```
Comment (juippis) [inline media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild:30]:
```
CMP: media-video/amdgpu-pro-amf-1.4.30.1620044/image with media-video/amdgpu-pro-amf-1.4.31.1654522/image
 FILES:-usr/lib64/libamfrt64.so.1.4.30
 FILES:+usr/lib64/libamfrt64.so.1.4.31
   ABI: libamfrt64.so.1(64) func(+47,-14) vars(+6,-2) [BREAKING]
------> FILES(+1,-1) ABI(+53,-16,>B<)
```
Aww crap.

File: media-video/amdgpu-pro-amf/Manifest [hunk 2]
```diff
@@ -2,3 +2,4 @@ DIST amdgpu-pro-amf-1.4.24.1452059.deb 1307216 BLAKE2B 2ecae593da961e2ff86e807c0
...
 DIST amdgpu-pro-amf-1.4.29.1580631.deb 1653852 BLAKE2B 5af5250c07e1b99ae8c1ee28507927e28ee7811929c789b5939a06f631b81155d6795e1392830945bf377f5298a5153e9af3541b2d94533661db394a059fb715 SHA512 07568c5e2f92f10274f7ac25a9d68539b8ce2e5d24c87e9b6a3aec2b8d5a32980c8fd802394dfa6eb421d3321dd60e07929e7f481578d23afcf623c5c9aa2fcc
 DIST amdgpu-pro-amf-1.4.30.1620044.deb 1671392 BLAKE2B 89611307dff4f77099ed52e8f7b523e842691b2fe970da7669a961048d5addd969068366a23d7c2df5710b67aeaf7a9cc3f4b902a708a4c686f00c32aa8cdb0c SHA512 4db5a814129a4d6e75466fd4e618eda155e031b70f184d5bf4dae78a13a61559fa26a1dcbc3d259064063467441822a34873af1142ab0366e38b61521c912bf4
+DIST amdgpu-pro-amf-1.4.31.1654522.deb 1694280 BLAKE2B ed2b6ab43b72bc7062bf7c5ec572983a9e3ebc96d2ed47d57903ebe12180a53cbc4848b81c9061da0aed30f32ec890b65cb75e234c1e942de6a277a5335d38bc SHA512 fdfc989f7dfcbe82081846b8cecd230fc927b1036a3cd5305e5299d843a4b3f19790f0f3be625322cb5cd79e2bf46721787843a52573baf65809196d6680f0f9
```
Comment (juippis) [inline media-video/amdgpu-pro-amf/Manifest:5]: I think we can safely drop the last part of this `${PV}` as there doesn't seem to be bumps between the minor versions.

File: media-libs/amdgpu-pro-vulkan/Manifest [hunk 3]
```diff
@@ -8,3 +8,5 @@ DIST amdgpu-pro-vulkan-22.40.6.1580631-amd64.deb 26403020 BLAKE2B 9af314b75ec6af
...
 DIST amdgpu-pro-vulkan-23.10.3.1620044-amd64.deb 28230396 BLAKE2B f0c59e5c955640ff8b98d1fe5982593b8ccd450fd51b11de256e30388afd7ccf7f8cf236149556186fb5b2bbbc60ff4c31a33aaf8456391403adb899dd77fc10 SHA512 7d454178c3ee601ebd51cbfa0ce72e8a097014a65079d5955702de030dc4172a4fdb368d40c781a54891bc2b31966455802db497237177112e35aab2d94c717b
 DIST amdgpu-pro-vulkan-23.10.3.1620044-i386.deb 29943120 BLAKE2B cb15cd340854d2d023c1d5d700790d7ef37f3cddb6ec5e11a0642da8dd6c9bcf375bcc3793f4a6a1e379c78b518c68e6bb188b91cd815bca1abc1e09f6516c02 SHA512 2c251b68a4910e8873256316248d4c932483e928e38b9752fd919df0165e3db134e7d08c2c1f95cceda24e47cd5b419d482243ff7e42af3cb4bec8abb4162436
+DIST amdgpu-pro-vulkan-23.20.0.1654522-amd64.deb 24480404 BLAKE2B 10e2dea7e03bc6c24b40b8d8c4933897e17edb8e9b1fb1aebc7a6151c2bf051bbd5e994b771a045e747730f41996b2b3819edd8163daeb22a0487d5a96ea15f7 SHA512 4014a3f5cdbc3e977d5953719e8d0ea34a0ff1cc9b7292513df69e98d1c6556643392dd77edf10a682f049c689d69c92ba307d5af5b8e68003ccbba014f7aa9a
```
Comment (juippis) [inline media-libs/amdgpu-pro-vulkan/Manifest:11]: Same `${PV}` thing here.

**Review by juippis — APPROVED** [review]
We should start removing some older versions, especially since nothing is stabilized cleaning up should be easy.

File: media-video/amdgpu-pro-amf/Manifest (hunk 2 above)
Comment (AdelKS) [reply media-video/amdgpu-pro-amf/Manifest:5]: Actually there is, but I simply keep the latest "patch" with the minor:
examples:
https://repo.radeon.com/amdgpu/5.5.2/ubuntu/pool/proprietary/a/amf-amdgpu-pro/
https://repo.radeon.com/amdgpu/5.5.3/ubuntu/pool/proprietary/a/amf-amdgpu-pro/

File: media-libs/amdgpu-pro-vulkan/Manifest (hunk 3 above)
Comment (AdelKS) [reply media-libs/amdgpu-pro-vulkan/Manifest:11]: Same here
https://repo.radeon.com/amdgpu/5.5.2/ubuntu/pool/proprietary/v/vulkan-amdgpu-pro/
https://repo.radeon.com/amdgpu/5.5.3/ubuntu/pool/proprietary/v/vulkan-amdgpu-pro/

File: media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild (hunk 1 above)
Comment (AdelKS) [reply media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild:30]: I should be binding `amdgpu-pro-amf` with their corresponding vulkan drivers ?

Comment (juippis) [reply media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild:30]: Sorry I don't understand the question. But this led me to do 
89c6f54d2280ce974f1dfa47389ec87f458a0176
e887ce7bd6dcb082de177e506935af71212c2912

Note that in future if the update introduces breaking SONAME changes, you'll have to bump the subslot value to match the SONAME value.

Comment (AdelKS) [reply media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.31.1654522.ebuild:30]: Sorry, basically `amdgpu-pro-amf` needs `amdgpu-pro-vulkan` to work, I wasn't sure how to interpret the code snippet you attached, looked like an API or ABI change ? so probably `amdgpu-pro-amf` needs to depend on a specific version of `amdgpu-pro-vulkan` ? 

ffmpeg depends only on `amf-headers` when it gets built, why would it need to be rebuilt with changing versions of `amdgpu-pro-amf` (binary only) ?

**Pull request closed without merging**